frontend/src/pages/projects/screens/detail/notebooks/types.ts

```diff
@@ -0,0 +1,42 @@
+import { ImageVersionDependencyType } from '~/pages/projects/screens/spawner/types';
+import { ImageStreamKind, ImageStreamSpecTagType } from '~/k8sTypes';
+import { NotebookImageAvailability } from './const';
+
+export type NotebookImage =
+  | {
+      imageAvailability: Exclude<NotebookImageAvailability, NotebookImageAvailability.DELETED>;
+      imageDisplayName: string;
+      tagSoftware: string;
+      dependencies: ImageVersionDependencyType[];
+    }
+  | {
+      imageAvailability: NotebookImageAvailability.DELETED;
+      imageDisplayName?: string;
+    };
+
+export type ImageData = {
+  imageStream: ImageStreamKind;
+  imageVersion: ImageStreamSpecTagType;
+};
+
+export type NotebookImageData =
+  | [data: null, loaded: false, loadError: undefined]
+  | [data: null, loaded: true, loadError: Error]
```

Comment:
I see that this traces down to `useImageSteams` which sets `loaded` to `true` along with an error. So the logic around this tyoe would work but I believe the pattern is to typically return an error with any value of `loaded`. `useFetchState` can return a subsequent error on refresh while `loaded` was previously set to true.
```suggestion
  | [data: null, loaded: false, loadError: Error | undefined]
```